backend/src/middleware/register.middleware.ts

```diff
@@ -0,0 +1,46 @@
+import { NextFunction, Request, Response } from 'express';
+import { sendLoginMail } from '../controllers/mailer.js';
+import { RequestBody } from './middleware.interface';
+import { userModel } from '../models/user.schema.js';
+
+export const store : Map<string, number> = new Map();
+
+export function challengeRegisterMiddleware(request : Request, response : Response, next : NextFunction) : void
+{
+	const { email, password, challenge } = (request.body as RequestBody);
+
+	userModel.findOne(
+		{
+			email
+		})
+		.then(result =>
+		{
+			if (result)
+			{
+				response.status(401).send();
+			}
+			else
+			{
+				if (email && email.includes('@') && password && password.length > 5 && challenge && store.get(email) === Number(challenge))
```

Comment:
We should have a helper file with validateEmail() and validatePassword() that is used for register and login middlewares